### PR TITLE
mgmtsystem_kpi rename to kpi

### PR DESCRIPTION
**Pull request by naousse**

### CODE REVIEW

File: kpi/README.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+When the action is created, a notification is sent to the person responsible
+for the action.
+
+When the action plan is reviewed (comments) and approved, each action of the
+plan is opened.
+
+When all actions of the plan are done, their effectiveness must be evaluated
+before closing the NC.
+
```
Comment (max3903) [inline kpi/README.rst:36]: @naousse Replace this usage section (from the nonconformity module) with the link to the video I sent you.

File: kpi/README.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/128/9.0
+
+Known issues / Roadmap
+======================
+
+* The custom emails should be replaced by Mail Tracking features and Subtypes (like in Project Tasks and Project Issues)
+* Automatically add responsible_user_id._uid, manager_user_id._uid, author_user_id._uid to chatter
```
Comment (max3903) [inline kpi/README.rst]: @naousse Wrong content here. Please add the remaining work to be done with the color stuff.

File: kpi/__openerp__.py [hunk 3]
```diff
@@ -20,42 +20,26 @@
...
         'security/ir.model.access.csv',
         'security/mgmtsystem_kpi_security.xml',
-        'mgmtsystem_kpi_view.xml',
+        'views/mgmtsystem_kpi_category.xml',
+        'views/mgmtsystem_kpi_history.xml',
+        'views/mgmtsystem_kpi_threshold_range.xml',
+        'views/mgmtsystem_kpi_threshold.xml',
+        'views/mgmtsystem_kpi.xml',
+        'views/menu.xml',
+        'data/mgmtsystem_kpi.xml',
```
Comment (max3903) [inline kpi/__openerp__.py]: @naousse Please remove `mgmtsystem_` from every file names.

**Comment by naousse** [comment]
**base.group_hr_user** is the employee user group i have and it is defined here:
**/home/odoo/odoo-dev/odoo/addons/hr/security/hr_security.xml.**
Without the hr dependance the system output that id is missed and when i add the hr dependance, everything works fine.

**Comment by max3903** [comment]
@naousse This is the wrong group. The `Employee` group (base.group_user) is created here:
https://github.com/odoo/odoo/blob/9.0/openerp/addons/base/security/base_security.xml#L24

**Comment by naousse** [comment]
Everything is ok now

**Comment by max3903** [comment]
@naousse Please review: https://github.com/naousse/server-tools/pull/1

**Comment by naousse** [comment]
:+1:  
I haven't paid attention about that

**Comment by max3903** [comment]
- [x] Depends on #513

**Comment by max3903** [comment]
@naousse Can you rebase please?

**Comment by max3903** [comment]
@naousse Can you rebase again please? Runbot should be usable now.

**Comment by max3903** [comment]
@naousse Few UI changes:

# KPI
- [x] Hide the company field on the list and form view. Should be displayed to multi_company group only.

# Thresholds
- [x] Hide the company field on the list and form view. Should be displayed to multi_company group only.
- [x] Move the company field on the same line as the name field
- [x] Add the label for the name
- [x] Hide the company field on the list view of Ranges and KPIs. Should be displayed to multi_company group only.

# Ranges
- [x] Hide the company field on the list and form view. Should be displayed to multi_company group only.
- [x] Add the label for the name and color
- [x] Name and company field should be on the same line
- [x] Type of min and max takes the whole line. Half of it is fine.
- [x] Fixed value and the datasource should be on the same line as the type.
- [x] Hide the company field on the list view of Thresholds. Should be displayed to multi_company group only.

**Comment by max3903** [comment]
@naousse There is one error on Runbot:
`openerp.addons.base.ir.ir_cron:139 _callback Model 'mgmtsystem.kpi' does not exist.`

# KPI History
- [x]  Hide the company field on the list and form view. Should be displayed to multi_company group only.

# Ranges
- [x] The minimum and maximum columns are empty on the list view.

**Review by max3903 — APPROVED** [review]
:+1:

File: kpi/README.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+* Use web_widget_color to display color associated to threshold range
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/Management-system/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+server-tools/issues/new?body=module:%20
```
Comment (elicoidal) [inline kpi/README.rst]: Remove this line following newest templates

File: kpi/README.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+A threshold is a list of ranges and a range is:
+
+* a name (like Good, Warning, Bad)
+* a minimum value (fixed, sql query or python code)
+* a maximum value (fixed, sql query or python code)
+* color (RGB code like #00FF00 for green, #FFA500 for orange, #FF0000 for red)
+
+
+Installation
```
Comment (elicoidal) [inline kpi/README.rst]: remove this section

File: kpi/README.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline kpi/README.rst]: https

File: kpi/__openerp__.py [hunk 7]
```diff
@@ -65,6 +48,5 @@
     ],
     "demo": [],
```
Comment (elicoidal) [inline kpi/__openerp__.py]: remove these 2 keys

File: kpi/models/kpi.py [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+            threshold_obj = obj.threshold_id
+            values = {
+                'kpi_id': obj.id,
+                'value': kpi_value,
+                'color': threshold_obj.get_color(kpi_value),
+            }
+            history_obj = self.env['kpi.history']
+            history_obj.create(values)
+            # obj.history_ids = history_obj.search([("kpi_id", "=", obj.id)])
```
Comment (elicoidal) [inline kpi/models/kpi.py]: remove dead code

File: kpi/models/kpi.py [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+                DATETIME_FORMAT)),
+            ('next_execution_date', '=', False),
+        ]
+        if 'filters' in self.env.context:
+            filters.extend(self.env.context['filters'])
+        obj_ids = self.search(filters)
+        res = None
+
+        try:
+            # res = self.compute_kpi_value()
```
Comment (elicoidal) [inline kpi/models/kpi.py]: remove dead code

File: kpi/models/kpi_category.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline kpi/models/kpi_category.py:1]: Can you use short headers

File: kpi/models/kpi_threshold.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        return result
+
+    @api.multi
+    def _compute_generate_invalid_message(self):
+        result = {}
+        for obj in self:
+            if obj.valid:
+                result[obj.id] = ""
+            else:
+                result[obj.id] = ("2 of your ranges are overlapping! Please "
```
Comment (elicoidal) [inline kpi/models/kpi_threshold.py]: s/2/Two

**Review by elicoidal — CHANGES_REQUESTED** [review]

File: kpi/models/kpi_threshold.py (hunk 11 above)
Comment (naousse) [reply kpi/models/kpi_threshold.py]: @elicoidal I made changes base on your review. You can check if everything is correct. Please I don't understand what you mean here

Comment (elicoidal) [reply kpi/models/kpi_threshold.py]: @naousse In English (and in French too actually), you normally do not start a sentence with a number but number writen in plain English hence my suggestion to change "2"  by "Two".

**Comment by max3903** [comment]
@elicoidal Can you update your review please?

**Comment by max3903** [comment]
Ping @dreispt

File: kpi/models/__init__.py [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline kpi/models/__init__.py:1]: Don't you need copyright notice?

File: kpi/models/kpi.py [hunk 13]
```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline kpi/models/kpi.py:1]: Don't you need copyright notice?

File: kpi/models/kpi_category.py (hunk 10 above)
Comment (elicoidal) [reply kpi/models/kpi_category.py:1]: Sorry: there must have been a misunderstanding: can you use short headers similar to https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L1 including your actual information?

File: kpi/views/kpi.xml [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+            </field>
+        </record>
+
+        <record id="view_kpi_filter" model="ir.ui.view">
+            <field name="name">kpi.filter</field>
+            <field name="model">kpi</field>
+            <field name="arch" type="xml">
+                <search string="KPI">
+                    <group>
+                        <filter name="active" icon="terp-document-new" domain="[('active','=',True)]" string="Active" help="Only active KPIs are computed by the scheduler based on the periodicity configuration."/>
```
Comment (elicoidal) [inline kpi/views/kpi.xml]: Can you try to improve the format of the line for readability (like above one)

**Review by elicoidal — COMMENTED** [review]
I think there has been a misunderstanding about the headers. Please check the example.
Note that you should have copyright notices in xml files as well (check: https://github.com/OCA/maintainer-tools/blob/master/template/module/data/model_name_data.xml#L2)

**Comment by max3903** [comment]
@naousse Can you fix based on @elicoidal latest comments? Thanks!

File: kpi/__init__.py [hunk 15]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# Copyright <YEAR(S)> Daniel Reis <dreis.pt@hotmail.com>
+# Copyright <YEAR(S)> Glen Dromgoole <gdromgoole@tier1engineering.com>
+# Copyright <YEAR(S)> Loic Lacroix <loic.lacroix@savoirfairelinux.com>
+# Copyright <YEAR(S)> Sandy Carter <sandy.carter@savoirfairelinux.com>
+# Copyright <YEAR(S)>Gervais Naoussi <gervaisnaoussi@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (dreispt) [inline kpi/__init__.py:3]: IMO copyright is attributed to original authors only, and contributors should not be included on authorship and copyright attribution.

**Review by dreispt — APPROVED** [review]
LGTM

**Comment by max3903** [comment]
@elicoidal Can you update your review? Copyrights headers are correct now.

File: kpi/__init__.py [hunk 16]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright <YEAR(S)> Savoir-faire Linux <https://www.savoirfairelinux.com/>
```
Comment (elicoidal) [inline kpi/__init__.py]: Please update the year

File: kpi/models/kpi_threshold.py [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+        if data.get('range_ids'):
+            for range1 in data['range_ids'][0][2]:
+                range_obj1 = range_obj1.browse(range1)
+                for range2 in data['range_ids'][0][2]:
+                    range_obj2 = range_obj2.browse(range2)
+                    if (range_obj1.valid and range_obj2.valid and
+                            range_obj1.min_value < range_obj2.min_value):
+                        if range_obj1.max_value > range_obj2.min_value:
+                            raise exceptions.Warning(
+                                _("Two of your ranges are overlapping!"),
```
Comment (elicoidal) [inline kpi/models/kpi_threshold.py]: Avoid "!" in error message, it makes users uncomfortable

**Comment by elicoidal** [comment]
Please fix the copyright then :+1:  on my side

**Comment by naousse** [comment]
@max3903 Please can you provide the right year for the copyright?

**Comment by max3903** [comment]
@naousse `2012 - Now`

**Comment by dreispt** [comment]
@elicoidal fix request was done - merging!
